### PR TITLE
[Release] Soft Launch - Add content security headers #165

### PR DESCRIPTION
**Pull request by daine**
# Summary | Résumé

This change adds content security headers and minor code changes to accommodate the security headers

## ⚠️ This PR can be tested through the following URL(s):
Main - https://staging.learning-resources.cdssandbox.xyz/
EN - https://en.staging.learning-resources.cdssandbox.xyz/
FR - https://fr.staging.learning-resources.cdssandbox.xyz/fr/

# 🚀  Release Checklist

Item | Checked / Reviewed by | Status | Notes
-- | -- | -- | --
Metrics - Google Analytics working and collecting data |  @daine @hillaryl   | Not started | [Prioritized for next release](https://github.com/cds-snc/resources-ressources/issues/154)
Content - Contentful |   | Not started |  
Content - English | @anikbrazeau  | Not started |  
Content - French | @anikbrazeau  | Not started | 
Infra - Domains OK | @daine   | ☀️ Ready |  
Infra - Amplify OK | @daine | ☀️ Ready  |  
Monitoring (Uptime Robot)| @daine | ☀️ Ready  |  [Status page dashboard](https://stats.uptimerobot.com/gP5gohqEVQ)
Accessibility - aXe DevTools | @daine @dylanzheng94   | Not started |  
Accessibility - Lighthouse checks | @daine @dylanzheng94 | Not started |  
Accessibility - WCAG 2.0 AA | @emanelfy | Not started | The intention is to achieve AAA for full launch
Infra - Content Security Policy | @daine  | ☀️ Ready |  

# After merge, do this:
From your terminal, make sure you are in the `main` branch, then type 
```bash
git tag -a v0.2.0 -m <release notes>
git push --tags
```
This bumps up the release version

### CODE REVIEW

**Comment by anikbrazeau** [comment]
I'm not sure I'm best placed to review/test this PR? What do you think @dylanzheng94?

Does this fix https://github.com/cds-snc/resources-ressources/issues/53? 🤔 
Are the accessibility checks failing a concern? 👀

**Review by dylanzheng94 — APPROVED** [review]

File: app/components/ResourceListItem.vue [hunk 1]
```diff
@@ -2,8 +2,7 @@
 
 <template>
   <div
-    class="py-5 px-1.5 rounded-2xl mb-4 hover:bg-gray-100 cursor-pointer drop-shadow-2xl -mx-2"
-    style="width: fit-content"
+    class="py-5 px-1.5 rounded-2xl mb-4 hover:bg-gray-100 cursor-pointer drop-shadow-2xl -mx-2 w-fit"
```
Comment (dylanzheng94) [inline app/components/ResourceListItem.vue:5]: `w-fit` doesn't work as intended

File: app/components/Footer.vue [hunk 2]
```diff
@@ -7,16 +7,14 @@
         <div class="flex flex-col w-auto">
           <nuxt-link
             :to="localePath({ name: 'about' })"
-            class="text-gray-800 text-2xl font-medium hover:underline hover:text-blue-700 mb-1"
-            style="width: fit-content"
+            class="text-gray-800 text-2xl font-medium hover:underline hover:text-blue-700 mb-1 w-fit"
```
Comment (dylanzheng94) [inline app/components/Footer.vue:10]: `w-fit` doesn't work as intended

**Review by dylanzheng94 — CHANGES_REQUESTED** [review]
`w-fit` doesn't actually perform `style="width: fit-content"`
another solution is required

**Review by dylanzheng94 — APPROVED** [review]

**Comment by daine** [comment]
> `w-fit` doesn't actually perform `style="width: fit-content"` another solution is required

Thanks for flagging this! The [Content Security Policy for styles](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src) doesn't like inline styles, so we might have to look further into this. I'm not seeing any errors right now on https://learning-resources.cdssandbox.xyz/ but I'll check our CSP report this week

**Pull request merged**